_posts/2025-05-11-running-hardware-in-the-loop-simulations-using-subnero-modems.md

```diff
@@ -0,0 +1,185 @@
+- Run real world network software with simulated acoustic links.
+
+If your modem is software-defined and open-architecture, such as Subnero's modems, HIL offers a development experience that closely mirrors actual deployment.
+
+## The Foundation: UnetStack v5 + Virtual Acoustic Ocean
+
+Modems running UnetStack v5 — such as those offered by Subnero — are fully software-defined and follow an open architecture. This enables users to customize physical and network layers, extend built-in components, and use the exact same software stack in both simulated and real-world deployments.
+
+[Virtual Acoustic Ocean (VAO)](https://github.com/org-arl/VirtualAcousticOcean.jl) is an open-source acoustic channel simulator. It supports standard propagation models, and allows users to define node positions, protocol stack parameters, and environmental conditions. Simulations can be run in interactive mode, making VAO suitable for everything from basic tests to complex protocol validation.
+
+With a modem running UnetStack v5 and VAO, users are now able to unlock scalable, reproducible HIL simulations that can closely mirror real-world underwater communication scenarios while enabling rapid, repeatable testing without going to sea.
+
+## Typical Use Cases
+
+This simulation setup supports a variety of scenarios. Researchers can use it to develop and validate new communication algorithms such as those at the physical or network layers, under realistic acoustic conditions. Commercial teams can evaluate performance for AUV-based data harvesting or remote sensing missions before heading to the field. Defense teams can rehearse complex mission scenarios virtually, especially in environments where access or testing is restricted.
+
+## Step-by-Step Guide to HIL Simulation
+
+Before beginning the simulation, ensure you have the following setup:
+
+- Two Subnero M25M series modems running UnetStack v5.
+- A laptop or mini-PC with Julia installed, capable of running the Virtual Acoustic Ocean (VAO) simulator.
+- An Ethernet switch to connect all devices to the same network.
+
+All components — modems and the VAO-running machine — should be on the same subnet to be able to reach each other using their IP address.
+
+> **TIP:** Use an isolated network switch (without internet access) to avoid interference from unrelated network traffic during simulation.
+
+> Note: This example uses Subnero M25M (MF-band) modems, but the same steps apply to Subnero's LF and HF modems. The simulation script provided is just one example — please refer to the VAO GitHub repository for the latest examples, which may differ. This guide shows how to connect two Subnero modems, but more nodes can be added by extending the script with additional addnode! lines.
+
+---
+
+### 1. Install VAO and Packages
+
+At a Julia terminal:
+
+```julia
+using Pkg
+Pkg.add("VirtualAcousticOcean")
+Pkg.add("UnderwaterAcoustics")
+Pkg.add("Sockets")
```

Comment:
We might want to add version specifiers here:

```julia
Pkg.add(name = "VirtualAcousticOcean", version = "0.6")
...

```

@veronicaSubnero had a brief moment of confusion on Friday because she had old versions of these packages installed, and then the code snippets below didn't work. 

Alternatively, we could complement `2-node-network.jl` with a `Project.toml` file and ask users to download both.

Reply:
Added the version for now.